### PR TITLE
Feat:  Docs

### PR DESCRIPTION
**Pull request by mykalmax**

### CODE REVIEW

File: web/server/models.py [hunk 1]
```diff
@@ -173,13 +174,36 @@ class Column(BaseModel):
...
+    owner: t.Optional[str] = None
+    kind: t.Optional[str] = None
+    batch_size: t.Optional[int] = None
+    cron: t.Optional[str] = None
+    stamp: t.Optional[str] = None
+    start: t.Optional[str] = None
+    retention: t.Optional[str] = None
+    storage_format: t.Optional[str] = None
+    time_column: t.Optional[str] = None
+    tags: t.Optional[str] = None
```
Comment (vchan) [inline web/server/models.py:187]: Should we keep this as a list of strings so it's more flexible?

File: web/server/api/endpoints/models.py [hunk 2]
```diff
@@ -18,15 +24,43 @@ def get_models(
...
+            details=models.ModelDetails(
+                owner=model.owner,
+                kind=model.kind.name,
+                batch_size=model.batch_size,
+                cron=model.cron,
+                stamp=model.stamp,
+                start=model.start,
+                retention=model.retention,
+                storage_format=model.storage_format,
+                time_column=f"{model.time_column.column} | {model.time_column.format}"
```
Comment (vchan) [inline web/server/api/endpoints/models.py:43]: These inline arguments are a little difficult to read. Can you please move them into variables, i.e. time_column and type?

File: web/client/src/library/pages/docs/SourceList.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      <ul className="p-2 overflow-auto scrollbar scrollbar--horizontal scrollbar--vertical">
+        {isArrayEmpty(modelsFiltered) && (
+          <li
+            key="not-found"
+            className="p-2"
+            onClick={() => {
+              setFilter('')
+            }}
+          >
+            No results found
```
Comment (vchan) [inline web/client/src/library/pages/docs/SourceList.tsx:48]: Do you usually capitalize all the words?

File: web/client/src/library/pages/docs/Search.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+          (max < index.length ? ' ...' : ''),
+      ])
+    }
+
+    return acc
+  }, [])
+}
+
+function createIndecies(models: Map<string, Model>): Array<[Model, string]> {
+  const indecies: Array<[Model, string]> = []
```
Comment (vchan) [inline web/client/src/library/pages/docs/Search.tsx:115]:
```suggestion
  const indices: Array<[Model, string]> = []
```

File: web/client/src/library/components/button/Button.tsx [hunk 5]
```diff
@@ -134,6 +138,22 @@ function ButtonPlain(
   )
 }
 
+function ButtoLinkPlain(
```
Comment (vchan) [inline web/client/src/library/components/button/Button.tsx:141]:
```suggestion
function ButtonLinkPlain(
```

File: web/client/src/library/components/documentation/Documantation.tsx
Comment (vchan) [inline web/client/src/library/components/documentation/Documantation.tsx:1]: Typo in the file name: Documentation.tsx

File: web/server/models.py (hunk 1 above)
Comment (mykalmax) [reply web/server/models.py:187]: i made it string so UI does not need to do a transformation

File: web/server/api/endpoints/models.py (hunk 2 above)
Comment (mykalmax) [reply web/server/api/endpoints/models.py:43]: sorry , did not get it , move where?

Comment (vchan) [reply web/server/api/endpoints/models.py:43]: Something like
```python
response = []
for model in context.models.values():
    time_column = f"{model.time_column.column} | {model.time_column.format}" if model.time_column else None
    details = models.ModelDetails(
        ...,
        time_column=time_column,
    )
    response.append(
        models.Model(
            name=model.name,
            ...,
            details=details,
    )
return response
```

File: web/client/src/library/pages/docs/Search.tsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+export default function Search({
+  models,
+  search,
+  setSearch,
+}: {
+  models: Map<string, Model>
+  search: string
+  setSearch: (search: string) => void
+}): JSX.Element {
+  const indeces = useMemo(() => createIndeces(models), [models])
```
Comment (vchan) [inline web/client/src/library/pages/docs/Search.tsx:18]:
```suggestion
  const indices = useMemo(() => createIndices(models), [models])
```

File: web/client/src/library/pages/docs/Search.tsx [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+          (max < index.length ? ' ...' : ''),
+      ])
+    }
+
+    return acc
+  }, [])
+}
+
+function createIndeces(models: Map<string, Model>): Array<[Model, string]> {
+  const indeces: Array<[Model, string]> = []
```
Comment (vchan) [inline web/client/src/library/pages/docs/Search.tsx:115]:
```suggestion
  const indices: Array<[Model, string]> = []
```

File: tests/web/test_main.py [hunk 8]
```diff
@@ -438,30 +437,30 @@ def test_fetchdf(web_sushi_context: Context) -> None:
...
-                )
-                for name, data_type in model.columns_to_types.items()
-            ],
-        ).dict()
-        for model in web_sushi_context.models.values()
-    ]
-
-    assert response.status_code == 200
-    assert json_models == response.json()
+# TODO: add better tests for this endpoint
```
Comment (vchan) [inline tests/web/test_main.py:440]: I think we can simplify this test. There's not a lot of complex logic in this endpoint, mostly just serializing fields. Maybe update this test to check the length of the payload and a serialized model (could just be the first model in the array) contains important fields like details, name, columns, etc. and whatever else you think should be tested, like maybe model type is correct?

**Review by vchan — APPROVED** [review]
Looking good. Just a few typos and add back in the `models` endpoint test.

File: web/server/api/endpoints/models.py
Comment (vchan) [inline web/server/api/endpoints/models.py:1]: I think you need the following line at the top of this file for linter to pass.
```python
from __future__ import annotations
```

**Pull request merged**